### ISSUE TITLE
Number Decimal Separator Incorrect for exponentially small numbers

### ISSUE THREAD

**Issue by evanliomain** [angular/angular.js#10342]
**Issue**

The number decimal separator are incorrect for the fr-fr locale in formatNumber for exponentially small numbers.

Expected

 - Decimal: ,

Actual

 - Decimal: .

In fact, it's an issue for all languages that do not have a dot as decimal separator.

I suggest the same use case than ["should filter exponentially small numbers"](https://github.com/angular/angular.js/blob/g3_v1_3/test/ng/filter/filtersSpec.js#L196) to test this issue:

```javascript
     it('should filter exponentially small numbers on non dot decimal separator languages', function() {
       $locale.NUMBER_FORMATS.DECIMAL_SEP = ',';
       expect(number(1e-50, 0)).toEqual('0');
       expect(number(1e-6, 6)).toEqual('0,000001');
       expect(number(1e-7, 6)).toEqual('0,000000');
 
       expect(number(-1e-50, 0)).toEqual('0');
       expect(number(-1e-6, 6)).toEqual('-0,000001');
       expect(number(-1e-7, 6)).toEqual('-0,000000');
     });
```

**Solve suggestion**

I figure out where the issue is localized: https://github.com/angular/angular.js/blob/g3_v1_3/src/ng/filter/filters.js#L206

```javascript
    if (fractionSize > 0 && number < 1) {
      formatedText = number.toFixed(fractionSize);
      number = parseFloat(formatedText);
    }
```

The [`.toFixed`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed) method is directly use to render the final number. And so, does NOT care about i18n at all.

To solve this issue, I suggest just to replace the dot by the expected decimal separator. As we deal this just number < 1, we don't have to manage thousand group separators.

```javascript
    if (fractionSize > 0 && number < 1) {
      formatedText = number.toFixed(fractionSize);
      number = parseFloat(formatedText);
      formatedText = formatedText.replace(DECIMAL_SEP, decimalSep);
    }
```

**Comment by Heddy147** [comment]
I have a similar problem.
There is a validation in a form with a field with type number.
See my Plunker demonstration:
http://plnkr.co/edit/OH8rQSkItepd7sEDVYeM?p=preview

I get an error if I type a number smaller than 0.000001

**Comment by petebacondarwin** [comment]
I have a fix for this if anyone cares to review: #12850
